shared/AppInsightsCommon/src/Offline.ts

```diff
@@ -0,0 +1,195 @@
+import {
+    IUnloadHook, arrForEach, createUniqueNamespace, eventOff, eventOn, getDocument, getNavigator, getWindow, isNullOrUndefined, isUndefined,
+    mergeEvtNamespace
+} from "@microsoft/applicationinsights-core-js";
+
+/**
+ * this is the callback that will be called when the network status changes
+ * @param onlineState this is the current network running state
+ */
+export type OfflineCallback = (onlineState: IOfflineState) => void;
+
+/**
+ * This is the enum for the different network states current ai is experiencing
+ */
+export const enum eOfflineValue {
+    Unknown = 0,
+    Online = 1,
+    Offline = 2
+}
+/**
+ * This is the interface for the Offline state
+ * runtime state is retrieved from the browser state
+ * user state is set by the user
+*/
+export interface IOfflineState {
+    readonly isOnline: boolean;
+    readonly rState: eOfflineValue; // runtime state
+    readonly uState: eOfflineValue; // user state
+}
+
+export interface IOfflineListener {
+    isOnline: () => boolean;
+    isListening: () => boolean;
+    unload: () => void;
+    addListener(callback: OfflineCallback): IUnloadHook;
```

Comment:
nit: Lets be consistent with the definition, so to be the same as the others
`addListener: (callback: OfflineCallback) => IUnloadHook;`